File: src/gestorAplicacion/entrenador/EntrenadorRoket.java
package gestorAplicacion.entrenador;

import java.util.ArrayList;

import gestorAplicacion.pokemon.Pokemon;

/**
 * Se utiliza para crear objetos del tipo EntrenadorRoket que heredan de la 
 * clase Entrenador, en este caso no nos intereza serializar ya que siempre será el mismo
 * 
 * @author David Urrego 
 * @author Pablo Usuga
 */

public class EntrenadorRoket extends Entrenador {
	
	/*al heredar de la clase entrenador, tambien hereda sus atributos y metodos
	 *solo que para el caso, ya tiene un nombre y unas medallas por defecto
	 */
	
	final String Nombre;
	ArrayList<String> Medallas;
		
	public EntrenadorRoket(String Nombre, String Genero, String edad, ArrayList<Pokemon> ListaPokemon,
			Pokemon PokemonInicial, ArrayList<String> Medallas) {
		super(Nombre, Genero, edad, ListaPokemon, PokemonInicial, Medallas);
		
		this.Nombre = "James";
		this.Medallas = null;
		
	}
	
	public String toString(){
		//metodo toString
		return "Soy James me me mes, pelea conmigo";
	}
	
	

}

File: python/gestorAplicacion/entrenador/EntrenadorRoket.py
from gestorAplicacion.entrenador.Entrenador import Entrenador
from gestorAplicacion.pokemon.Pokemon import Pokemon
from gestorAplicacion.pokemon.Habilidad import Habilidad

class EntrenadorRoket(Entrenador):

    _entrenadoresRoket = []

    def __init__(self, Nombre, ListaPokemon, pokemonInicial,Medellas = None):
        super().__init__(Nombre, ListaPokemon, pokemonInicial,Medellas)
        if (isinstance(self,EntrenadorRoket) == True):
            EntrenadorRoket._entrenadoresRoket.append(self)

    def __str__(self):
        return "mi nombre es {} y quiero robar tu pokemon".format(self.getNombre())

    @classmethod
    def getEntrenadoresRoket(cls):
        return cls._entrenadoresRoket
